src/main/scala/org/broadinstitute/dsde/rawls/monitor/BootMonitors.scala

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsde.rawls
+
+import akka.actor.ActorRef
+import akka.pattern._
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, GraphContainerDAO, DataSource}
+import org.broadinstitute.dsde.rawls.jobexec.SubmissionSupervisor.SubmissionStarted
+import org.broadinstitute.dsde.rawls.model.{PendingBucketDeletions, WorkspaceName}
+import org.broadinstitute.dsde.rawls.monitor.BucketDeletionMonitor.DeleteBucket
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+// handles monitors which need to be started at boot time
+object BootMonitors {
+
+  def restartMonitors(dataSource: DataSource, containerDAO: GraphContainerDAO, gcsDAO: GoogleServicesDAO, submissionSupervisor: ActorRef, bucketDeletionMonitor: ActorRef) = {
+    startBucketDeletionMonitor(dataSource, containerDAO, bucketDeletionMonitor)
+    startSubmissionMonitor(dataSource, containerDAO, gcsDAO, submissionSupervisor)
+  }
+
+  private def startBucketDeletionMonitor(dataSource: DataSource, containerDAO: GraphContainerDAO, bucketDeletionMonitor: ActorRef) = {
+    dataSource.inTransaction() { txn =>
+      containerDAO.workspaceDAO.loadPendingBucketDeletions(txn) match {
+        case Some(pendingDeletions) => pendingDeletions.buckets foreach {
+          bucketDeletionMonitor ! DeleteBucket(_)
+        }
+        case None => containerDAO.workspaceDAO.savePendingBucketDeletions(PendingBucketDeletions(Set.empty), txn)
+      }
+    }
+  }
+
+  private def startSubmissionMonitor(dataSource: DataSource, containerDAO: GraphContainerDAO, gcsDAO: GoogleServicesDAO, submissionSupervisor: ActorRef) = {
+    dataSource.inTransaction() { txn =>
+      containerDAO.submissionDAO.listAllActiveSubmissions(txn) foreach { activeSub =>
+        val wsName = WorkspaceName(activeSub.workspaceNamespace, activeSub.workspaceName)
+        val submitter = activeSub.submission.submitter
+        val subId = activeSub.submission.submissionId
+
+        val subStartMessage = gcsDAO.getUserCredentials(submitter) map { credentialOpt =>
+          credentialOpt match {
+            case None => throw new RawlsException(s"Cannot start Submission Monitor because credentials were not retrieved for user ${submitter.userSubjectId.value}, submitter of ${subId}")
+            case Some(credential) => SubmissionStarted(wsName, subId, credential)
+          }
+        }
+
+        pipe(subStartMessage) to submissionSupervisor
```

Comment:
it is not equivalent because the throwable will not be handled anywhere. Instead of throwing the throwable how about just logging it. That is about all we would do anyway.